src/ring_theory/jacobson_ideal.lean

```diff
@@ -86,6 +97,110 @@ theorem mem_jacobson_iff {I : ideal R} {x : R} :
+begin
+  use λ hI, ⟨{J : ideal R | I ≤ J ∧ J.is_maximal}, ⟨λ _ hJ, or.inl hJ.right, hI.symm⟩⟩,
+  rintros ⟨M, hM, hInf⟩,
+  refine le_antisymm (λ x hx, _) le_jacobson,
+  rw [hInf, mem_Inf],
+  intros I hI,
+  cases hM I hI with is_max is_top,
+  { exact (mem_Inf.1 hx) ⟨le_Inf_iff.1 (le_of_eq hInf) I hI, is_max⟩ },
+  { exact is_top.symm ▸ submodule.mem_top }
+end
+
+/-- An ideal `I` equals its jacobson radical if and only if every element outside `I`
+also lies outside of a maximal ideal containing `I`. -/
+lemma eq_jacobson_iff_not_mem {I : ideal R} :
+  I.jacobson = I ↔ ∀ x ∉ I, ∃ M : ideal R, (I ≤ M ∧ M.is_maximal) ∧ x ∉ M :=
+begin
+  split,
+  { intros h x hx,
+    erw [← h, mem_Inf] at hx,
+    push_neg at hx,
+    exact hx },
+  { refine λ h, le_antisymm (λ x hx, _) le_jacobson,
+    contrapose hx,
+    erw mem_Inf,
+    push_neg,
+    exact h x hx }
+end
+
+theorem map_jacobson_of_surjective {f : R →+* S} (hf : function.surjective f) {I : ideal R} :
+  ring_hom.ker f ≤ I → map f (I.jacobson) = (map f I).jacobson :=
+begin
+  intro h,
+  rw [ideal.jacobson, ideal.jacobson],
+  have : ∀ J ∈ {J : ideal R | I ≤ J ∧ J.is_maximal}, f.ker ≤ J := λ J hJ, le_trans h hJ.left,
+  refine trans (map_Inf hf this) (le_antisymm _ _),
+  { refine Inf_le_Inf (λ J hJ, ⟨comap f J, ⟨⟨le_comap_of_map_le hJ.1, _⟩, map_comap_of_surjective f hf J⟩⟩),
+    haveI : J.is_maximal := hJ.right,
+    exact comap_is_maximal_of_surjective f hf },
+  { refine Inf_le_Inf' (λ j hj, hj.rec_on (λ J hJ, _)),
+    rw ← hJ.2,
+    cases map_eq_top_or_is_maximal_of_surjective f hf hJ.left.right with htop hmax,
+    { exact htop.symm ▸ set.mem_insert ⊤ _ },
+    { exact set.mem_insert_of_mem ⊤ ⟨map_mono hJ.1.1, hmax⟩ } },
+end
+
+theorem comap_jacobson_of_surjective {f : R →+* S} (hf : function.surjective f) {K : ideal S} :
+  comap f (K.jacobson) = (comap f K).jacobson :=
+begin
+  rw [ideal.jacobson, ideal.jacobson],
+  refine le_antisymm _ _,
+  { refine le_trans (comap_mono (le_of_eq (trans top_inf_eq.symm Inf_insert.symm))) _,
+    rw [comap_Inf', Inf_eq_infi],
+    refine infi_le_infi_of_subset (λ J hJ, _),
+    have : comap f (map f J) = J := trans (comap_map_of_surjective f hf J)
+      (le_antisymm (sup_le_iff.2 ⟨le_of_eq rfl, le_trans (comap_mono bot_le) hJ.left⟩) le_sup_left),
+    cases map_eq_top_or_is_maximal_of_surjective _ hf hJ.right with htop hmax,
+    { refine ⟨⊤, ⟨set.mem_insert ⊤ _, htop ▸ this⟩⟩ },
+    { refine ⟨map f J, ⟨set.mem_insert_of_mem _
+        ⟨le_map_of_comap_le_of_surjective f hf hJ.1, hmax⟩, this⟩⟩ } },
+  { rw comap_Inf,
+    refine le_infi_iff.2 (λ J, (le_infi_iff.2 (λ hJ, _))),
+    haveI : J.is_maximal := hJ.right,
+    refine Inf_le ⟨comap_mono hJ.left, comap_is_maximal_of_surjective _ hf⟩ }
+end
+
+/-- An ideal `I` of `R` is equal to its jacobson radical if and only if
+the jacobson radical of the quotient ring `R/I` is the zero ideal -/
+theorem jacobson_eq_iff_jacobson_bot_eq {I : ideal R} :
```

Comment:
The name of this theorem doesn't make clear that `bot` is an ideal of the quotient ring. The best I can come up with is:
```suggestion
theorem jacobson_eq_iff_jacobson_quotient_eq_bot {I : ideal R} :
```
but feel free to make your own suggestions.